lib/lighthouse/matchers/rspec.rb

```diff
@@ -2,28 +2,26 @@
 
 require 'rspec/expectations'
 require 'lighthouse/matchers'
+require 'lighthouse/audit_service'
+require 'lighthouse/message_service'
 require 'json'
 
 RSpec::Matchers.define :pass_lighthouse_audit do |audit, score: nil|
   match do |target|
-    score ||= Lighthouse::Matchers.minimum_score
-    port = Lighthouse::Matchers.remote_debugging_port
-    runner = Lighthouse::Matchers.runner
-
-    url = target.respond_to?(:current_url) ? target.current_url : target
-    opts = "'#{url}' --quiet --output=json #{"--port=#{port}" if port}".strip
-    cmd = Lighthouse::Matchers.lighthouse_cli
-    output = runner.call("#{cmd} #{opts}")
-    results = JSON.parse(output)
-    actual_score = results.dig('categories', audit.to_s, 'score') * 100
-    actual_score >= score
+    AuditService.new(url(target), audit, score(score)).passing_score?
   end
 
   failure_message do |target|
-    url = target.respond_to?(:current_url) ? target.current_url : target
-    <<~FAIL
-      expected #{url} to pass Lighthouse #{audit} audit
-      with a minimum score of #{score}
-    FAIL
+    MessageService.new(url(target), audit, score(score)).audit_fail_message
+  end
+
+  private
+
+  def url(target)
+    target.respond_to?(:current_url) ? target.current_url : target
+  end
+
+  def score(score)
```

Comment:
If my suggestion about moving this to the service above is acceptable, then this method could be removed

Reply:
See my above comment, I personally prefer to keep this here. Happy to discuss further though.

Reply:
If this does stay here, I wonder if it can just be a variable assignment just after line 9 for both `match` and `message` to use? It seems a little overkill for it to be a method since it's basically just an `||=`

Reply:
That makes sense. Will make that change.